module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -2,7 +2,9 @@ package heropool
 
 import (
 	"fmt"
+	"math/rand"
```

Comment:
In light of our [recent refactoring](https://github.com/onflow/flow-go/pull/4052) across the codebase, please use the `"github.com/onflow/flow-go/utils/rand"` instead of the `math/rand.` 

Please see the example [here](https://github.com/onflow/flow-go/pull/4052/files#diff-2e91123b8c596c937539bcc5aa53919f116b3207234cc83ed5b4d981493d6705R181).

Reply:
Understood. I have a first question. I found this in the package description -
```
// This package does not implement any determinstic RNG (Pseudo-RNG) based on
// user input seeds. For the deterministic use-cases please use `flow-go/crypto/random`.
```
So, I went to the package in question and found the following 
```
	// Store returns the internal state of the random generator.
	// The internal state can be used as a seed input for the function
	// Restore to restore an identical PRG (with the same internal state)
	Store() []byte
```
However, I couldn't find `Restore` function anywhere. I did git grep as usual, and I found 
```RestoreChacha20PRG``` in one of the tests.

So, looks like I need to log the output of the ```Store``` .
But, I am not 100% sure about it. Mainly, cause in both packages I don't see a clear way to restore the PRG.

So, if for example I would  log rand.Store() of  `flow-go/crypto/random` and later on use functions from 
 `flow-go/utils/random` and say the test fails. How shell I restore the RNG. There is no 'Restore' in the interface so I would need to cast to chachaPRG and use it's  ```RestoreChacha20PRG``` ? TBH I didn't give it yet a try, by if you know already the answer pls advise. If it needs some investigation from your side, then I can try to cast and see if I can somehow rerun the same test.

Reply:
We don't need a deterministic PRNG for this case. Please use `github.com/onflow/flow-go/utils/rand`

Reply:
@yhassanzadeh13  thank you for the suggestion. I have applied requested changes. I am not sure about 

```
randomIntN := func(length int) int {
	random, err := rand.Uintn(uint(length))
	require.Nil(t, err)
	return int(random)
}
```
may be one should not exit the test but try again ... and only then exit or even log error and pick some predefined number. But I guess err should almost never happen.

Reply:
@kozlovb thanks, yes, this is exactly as we need, to fail the test upon an error.